source/contexts/core/entities/Entity.gd

```diff
@@ -0,0 +1,18 @@
+class_name Entity
+
+extends Resource
+
+# read only, settable only within the owner script
+var name: String setget _set_name, _get_name
```

Comment:
Godot already has a `name` field, so I try not to call things `name`. Maybe `entity_name`? Also, I don't see any logic in the getter/setter, so maybe it can just be public?